### PR TITLE
Revert `operation.amount` to `uint128_t`

### PR DESCRIPTION
**Pull request by envin3**
Use uint_128 precision for amounts.

### CODE REVIEW

File: cpp/contracts/adapter.cpp [hunk 1]
```diff
@@ -53,7 +53,7 @@ void adapter::create(
       check(is_account(token), "token account does not exist");
    }
 
-   symbol non_local_token_symbol = symbol(symbol_code("XXX"), 0);
+   symbol non_local_token_symbol = symbol(symbol_code("XXX"), token_symbol.precision());
```
Comment (gitmp01) [inline cpp/contracts/adapter.cpp:56]: Interesting...

File: cpp/contracts/tables/adapter_registry.table.hpp [hunk 2]
```diff
@@ -23,8 +23,9 @@ namespace eosio {
    // WETH ERC20 => local chain is Ethereum, then adapter's registry is
    //
    // |_____token____|___token_symbol___|______token_bytes_______|_____xerc20_____|
-   // |      ''      |     '0,XXX'      | bytes32(address(WETH)) | 'xweth.token'  |
-   // NOTE: for not local token_symbol = '0,XXX'
+   // |      ''      |     '18,XXX'      | bytes32(address(WETH)) | 'xweth.token'  |
+   //
+   // NOTE: for not local token_symbol = '<actual_precision>,XXX'
```
Comment (gitmp01) [inline cpp/contracts/tables/adapter_registry.table.hpp:28]: Let's be a bit more explicit
```suggestion
   // NOTE: for not local token_symbol = '<origin-chain-precision>,XXX'
```

File: cpp/contracts/utils.hpp [hunk 3]
```diff
@@ -131,20 +131,20 @@ namespace eosio {
...
-         adjusted_amount = from_asset.amount * factor;
+         adjusted_amount = amount * factor;
       } else {
          print("WARNING: Operation precision exceeds destination symbol; amount will be floored to destination symbol precision.");
          factor = powint(10, exp);
-         adjusted_amount = from_asset.amount / factor;
+         adjusted_amount = amount / factor;
       }
-      return asset(adjusted_amount, to_sym);
+      return asset(adjusted_amount, to_symbol);
```
Comment (gitmp01) [inline cpp/contracts/utils.hpp:147]: Nice!

File: cpp/contracts/adapter.cpp [hunk 4]
```diff
@@ -189,8 +189,8 @@ void adapter::settle(const name& caller, const operation& operation, const metad
    adapter_registry_table registry_data = _registry.get();
    check(registry_data.token_bytes == operation.token, "underlying token does not match with adapter registry");
    // symbol check if underlying is EOS type
-   if (registry_data.token != name(0)) // TODO add test
-      check(registry_data.xerc20_symbol == operation.amount.symbol, "registered xerc20 symbols differs from the operation one");
+   // if (registry_data.token != name(0)) // TODO add test
+   //    check(registry_data.xerc20_symbol == operation.amount.symbol, "registered xerc20 symbols differs from the operation one");
```
Comment (gitmp01) [inline cpp/contracts/adapter.cpp:193]: Do we still need this?

File: cpp/test/adapter-non-local.test.js [hunk 5]
```diff
@@ -546,7 +547,7 @@ describe('Adapter EVM -> EOS testing', () => {
         [evmXERC20],
       )
 
-      expect(after[recipient][evmXERC20.symbol]).to.be.equal(operation.amount)
+      expect(after[recipient][evmXERC20.symbol]).to.be.equal(operationAsset.toString())
```
Comment (gitmp01) [inline cpp/test/adapter-non-local.test.js:550]: Put operation asset declaration closer here

**Review by gitmp01 — APPROVED** [review]
Minor suggestions, but it LGTM!

P.s. I liked the idea to put the origin chain precision, still not sure what the usage is going to be, but better this way as it can be configurable

File: cpp/contracts/adapter.cpp (hunk 4 above)
Comment (envin3) [reply cpp/contracts/adapter.cpp:193]: No, symbol in the amount is gone indeed.

File: cpp/contracts/adapter.cpp (hunk 1 above)
Comment (envin3) [reply cpp/contracts/adapter.cpp:56]: We need the precision also forn non local assets because we need to create the mint quantity asset.

**Pull request merged**